server/src/main/java/org/opensearch/extensions/rest/ExtensionRestRequest.java

```diff
@@ -301,6 +306,15 @@ public String getRequestIssuerIdentity() {
         return principalIdentifierToken;
     }
 
+    /**
+     * Assert that the principal identifier token is not null.
+     */
+    public void validateRequestIssuerIdentity() {
```

Comment:
I think this is not needed (after https://github.com/opensearch-project/OpenSearch/pull/8679/files#r1291337078): the `in.readString()` will refuse to read `null` string, so there should be no way for `null` to sneak in.